sdk/examples/devmode_rust/src/engine.rs

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use std::sync::mpsc::{Receiver, RecvTimeoutError};
+use std::time;
+
+use rand;
+use rand::Rng;
+
+use sawtooth_sdk::consensus::{engine::*, service::Service};
+
+pub struct DevmodeService {
+    pub service: Box<Service>,
+}
+
+impl DevmodeService {
+    pub fn new(service: Box<Service>) -> Self {
+        DevmodeService { service }
+    }
+
+    fn get_chain_head(&mut self) -> Block {
+        self.service
+            .get_chain_head()
+            .expect("Failed to get chain head")
+    }
+
+    fn get_block(&mut self, block_id: BlockId) -> Block {
```

Comment:
This should be a `&BlockId` param, since you're not consuming it.

Reply:
`Service.get_blocks` takes a vector of block ids, I think consuming them.